### PR TITLE
Jwt eli class exercise

### PR DESCRIPTION
**Pull request by RonAzar**

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    override fun filter(requestContext: ContainerRequestContext) {
+        if (requestContext.uriInfo.path == "jwt/users/login") return
+
+        // Verify the JWT token from the cookie
+        val cookie = requestContext.cookies
+        val jwtClaims = verify(cookie["ron_cookie_name"]?.value)
+
+        if (jwtClaims == null) {
+            requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).entity("Invalid or expired token").build())
+        }
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt:25]: handle this in the catch

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import com.hibob.academy.service.SessionService.Companion.SECRET_KEY
+import jakarta.ws.rs.core.Response
+import jakarta.ws.rs.ext.Provider
+
+
+@Component
+@Provider
+class AuthenticationFilter : ContainerRequestFilter {
+    override fun filter(requestContext: ContainerRequestContext) {
+        if (requestContext.uriInfo.path == "jwt/users/login") return
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt:17]: consider making "jwt/users/login" a const

File: src/main/kotlin/com/hibob/jwt/jwtClassExercise.kt
Comment (elijulian) [inline src/main/kotlin/com/hibob/jwt/jwtClassExercise.kt:1]: revert

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt (hunk 2 above)
Comment (elijulian) [reply src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt:17]: the cookie name should be a const too

File: src/main/kotlin/com/hibob/jwt/jwtClassExercise.kt
Comment (RonAzar) [reply src/main/kotlin/com/hibob/jwt/jwtClassExercise.kt:1]: Hey Eli, I don’t fully understand. Could you clarify where I need to revert?

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+        // Verify the JWT token from the cookie
+        val cookie = requestContext.cookies[COOKIE_NAME]?.value
+
+        try {
+            val jwtClaims = verify(cookie) ?: throw Exception("Invalid or expired token")  // If claims are null, throw an exception
+        } catch (e: Exception) {
+            // Catch any exceptions (including null or verification failure) and abort with UNAUTHORIZED status
+            requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).entity("Invalid or expired token").build())
+        }
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt:32]: you already have an exception being thrown in the verify method? no need to throw a new exception here. just handle the null case in the verify method

Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt:32]: Got it 
Thanks Eli!

**Review by elijulian — COMMENTED** [review]
please test this. based on what i see it will not reject calls that are not authenticated

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+        // Verify the JWT token from the cookie
+        val cookie = requestContext.cookies[COOKIE_NAME]?.value
+
+        try {
+            val jwtClaims = verify(cookie) ?: throw Exception("Invalid token")
+        } catch (e: Exception) {
+            // Catch any exceptions (including null or verification failure) and abort with UNAUTHORIZED status
+            requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).entity("Invalid or expired token").build())
+        }
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt:32]: i would prefer that all handling of the cookie be done in the verify method. That way this method is clean and doesn't need to know the details of the verification process.

File: src/main/kotlin/com/hibob/jwt/jwtClassExercise.kt
Comment (elijulian) [inline src/main/kotlin/com/hibob/jwt/jwtClassExercise.kt:1]: revert

File: src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt (hunk 4 above)
Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt]: Refactored authentication filter to move cookie handling into verify method for cleaner separation of concerns.

**Review by elijulian — APPROVED** [review]

**Pull request merged**